test/e2e/json-rpc/wallet_requestPermissions.spec.ts

```diff
@@ -1,58 +1,42 @@
-const { strict: assert } = require('assert');
-const {
-  defaultGanacheOptions,
-  withFixtures,
-  switchToNotificationWindow,
-  switchToOrOpenDapp,
-  unlockWallet,
-} = require('../helpers');
-const FixtureBuilder = require('../fixture-builder');
+import { strict as assert } from 'assert';
+import { withFixtures } from '../helpers';
+import FixtureBuilder from '../fixture-builder';
+import { loginWithBalanceValidation } from '../page-objects/flows/login.flow';
+import TestDapp from '../page-objects/pages/test-dapp';
 
 describe('wallet_requestPermissions', function () {
   it('executes a request permissions on eth_accounts event', async function () {
     await withFixtures(
       {
         dapp: true,
-        fixtures: new FixtureBuilder()
-          .withPermissionControllerConnectedToTestDapp()
-          .build(),
-        ganacheOptions: defaultGanacheOptions,
-        title: this.test.title,
+        fixtures: new FixtureBuilder().build(),
+        title: this.test?.title,
       },
       async ({ driver }) => {
-        await unlockWallet(driver);
+        await loginWithBalanceValidation(driver);
+        const testDapp = new TestDapp(driver);
+        await testDapp.openTestDappPage();
 
         // wallet_requestPermissions
-        await driver.openNewPage(`http://127.0.0.1:8080`);
-
         const requestPermissionsRequest = JSON.stringify({
           jsonrpc: '2.0',
           method: 'wallet_requestPermissions',
           params: [{ eth_accounts: {} }],
         });
-
         await driver.executeScript(
           `window.ethereum.request(${requestPermissionsRequest})`,
         );
 
-        await switchToNotificationWindow(driver);
-
-        await driver.clickElement({
-          text: 'Connect',
-          tag: 'button',
-        });
-
-        await switchToOrOpenDapp(driver);
+        // confirm connect account
+        await testDapp.confirmConnectAccountModal();
 
         const getPermissionsRequest = JSON.stringify({
           method: 'wallet_getPermissions',
         });
-
         const getPermissions = await driver.executeScript(
           `return window.ethereum.request(${getPermissionsRequest})`,
         );
-
-        assert.strictEqual(getPermissions[0].parentCapability, 'eth_accounts');
+        assert.strictEqual(getPermissions[1].parentCapability, 'eth_accounts');
```

Comment:
intended change because since we removed `withPermissionControllerConnectedToTestDapp` at the test setup, the permission order has changed

Reply:
make sense